packages/codebytes/src/editor.tsx

```diff
@@ -1,10 +1,164 @@
-import React from 'react';
+import {
+  FillButton,
+  FlexBox,
+  Spinner,
+  TextButton,
+  ToolTip,
+} from '@codecademy/gamut';
+import { CopyIcon } from '@codecademy/gamut-icons';
+import { theme } from '@codecademy/gamut-styles';
+import styled from '@emotion/styled';
+import React, { useState } from 'react';
+
+import type { languageOption } from './consts';
+import { Drawers } from './drawers';
+
+const Output = styled.pre<{ hasError: boolean }>`
+  width: 100%;
+  height: 20rem;
+  margin: 0;
+  padding: 0 1rem;
+  font-family: Monaco;
+  font-size: 0.875rem;
+  overflow: auto;
+  ${({ hasError }) => `
+  color: ${hasError ? theme.colors.orange : theme.colors.white};
+  background-color: ${theme.colors['gray-900']};
+`}
+`;
+
+const CopyIconStyled = styled(CopyIcon)`
+  margin-right: 0.5rem;
+`;
+
+const DOCKER_SIGTERM = 143;
 
 type EditorProps = {
+  hideCopyButton: boolean;
+  language: languageOption;
   text: string;
-  onChange: (text: string) => void;
+  // eslint-disable-next-line react/no-unused-prop-types
+  onChange: (
+    text: string
+  ) => void /* TODO: Add onChange behavior in DISC-353 */;
+  onCopy?: (text: string, language: string) => void;
 };
 
-export const Editor: React.FC<EditorProps> = ({ text, onChange }) => {
-  return <textarea value={text} onChange={(e) => onChange(e.target.value)} />;
+interface Response {
+  stderr: string;
+  stdout: string;
+  exit_code: number;
+}
+
+export const Editor: React.FC<EditorProps> = ({
+  language,
+  text,
+  hideCopyButton,
+  onCopy,
+}) => {
+  const [output, setOutput] = useState('');
+  const [status, setStatus] = useState<'ready' | 'waiting' | 'error'>('ready');
+  const [isCodeByteCopied, setIsCodeByteCopied] = useState(false);
+  const onCopyClick = () => {
+    if (!isCodeByteCopied) {
+      navigator.clipboard
+        .writeText(text)
+        // eslint-disable-next-line no-console
+        .catch(() => console.error('Failed to copy'));
+      onCopy?.(
+        text,
+        language
+      ); /* TODO: pass in onCopyBBCodeblock behavior from the future version we migrate to Next.js  */
+      setIsCodeByteCopied(true);
+    }
+  };
+
+  const setErrorStatusAndOutput = (message: string) => {
+    setOutput(message);
+    setStatus('error');
+  };
+
+  const handleSubmit = () => {
+    if (text.trim().length === 0) {
+      return;
+    }
+    const data = {
+      language,
+      code: text,
+    };
+    setStatus('waiting');
+    setOutput('');
+
+    const snippetsEndpoint =
+      'https://' + process.env.GATSBY_CONTAINER_API_BASE + '/snippets';
```

Comment:
i assume we'll need to handle this env var grab differently now

Reply:
yeah, this won't work here, but I can def update this, i think i understand the proper implementation for storybook now!

Reply:
@borisonr, i'm going to backtrack here and go with a placeholder endpoint for now in this pr, not expecting it to work. i think it's best to handle it in either its own pr or in the simple code editor pr, since i want to talk to infra and arik a bit more about the most secure way of mentioning this endpoint in an opensource repo. it may not be a big deal, but think it merits a conversation to put it in the right place.

Reply:
gotcha, as long as we're clear on which ticket will include this work, that sounds good to me!

Reply:
Touching base with Arik today, and it'll happen in the next ticket when we add monaco! I'll update the ticket description to be clear.